be/src/connector/file_connector.h

```diff
@@ -0,0 +1,77 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include "column/vectorized_fwd.h"
+#include "connector/connector.h"
+#include "exec/vectorized/file_scanner.h"
+
+namespace starrocks::connector {
+
+class FileConnector final : public Connector {
+public:
+    ~FileConnector() override = default;
+
+    DataSourceProviderPtr create_data_source_provider(starrocks::vectorized::ConnectorScanNode* scan_node,
+                                                      const TPlanNode& plan_node) const override;
+};
+
+class FileDataSource;
+class FileDataSourceProvider;
+
+class FileDataSourceProvider final : public DataSourceProvider {
+public:
+    ~FileDataSourceProvider() override = default;
+    friend class FileDataSource;
+    FileDataSourceProvider(vectorized::ConnectorScanNode* scan_node, const TPlanNode& plan_node);
+    DataSourcePtr create_data_source(const TScanRange& scan_range) override;
+
+    bool insert_local_exchange_operator() const override { return true; }
+    bool accept_empty_scan_ranges() const override { return false; }
+
+protected:
+    vectorized::ConnectorScanNode* _scan_node;
+    const TFileScanNode _file_scan_node;
+};
+
+class FileDataSource final : public DataSource {
```

Comment:
suggest to use independent io threads for file data source in _prepare_pipeline_driver()

Reply:
no no the beauty of this connector is to embrace all data sources without worrying threads and details

Reply:
Besides, connector is  running on scan threads apart from execution threads.

Reply:
Could the io thread be hang for some reason, such as the network

Reply:
But I think io thread is enough. Because if we following your logic, then we will have
- Hdfs IO thread
- JDBC IO thread
- MySQL IO thread
- ...
Because we don't want each component interfere each other right?

Reply:
I think we have multiple ways to set up independent io thread pools. For example, from a functional point of view, the above can be classified as external table io threads, or from the perspective of pull/push model, the above may be a pull model, but HTTP/Kafka may be the push model, because we need to wait for the data to arrive, instead of reading it directly.

Reply:
I'm ok to add indenpdent threads, but only if two pointe are met:
1. DataSource should not be aware of threads or execution units. 
2. There are a lot of threads. The new indepdenet threads size should be small.

Reply:
I'm afraid that would introduce too much threads for different `ScanOperator`. Maybe we need to unify these threads?